src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -91,36 +92,44 @@ function generate_gcno_files() {
 # Reads the list of arguments provided by the caller (using $@) and uses them
 # to produce an instrumented binary using g++.
 function generate_instrumented_binary() {
+  local path_to_binary="${1}"; shift
   # "-fprofile-arcs -ftest-coverage" tells the compiler to generate coverage
   # information needed by gcov and include additional code in the object files
   # for generating the profiling.
-  g++ -fprofile-arcs -ftest-coverage "$@" -o ./coverage_srcs/test && return 0
+  g++ -fprofile-arcs -ftest-coverage "$@" -o "$path_to_binary"  && return 0
   fail "Couldn't produce the instrumented binary for $@"
   return 1
 }
 
 # Execute the test coverage_srcs/test and generate the
 # coverage_srcs/test.gcda file.
 function generate_gcda_file() {
-  ./coverage_srcs/test && return 0
-  fail "Couldn't execute the instrumented binary for $@"
+  local path_to_binary="${1}"
+  "$path_to_binary" && return 0
+  fail "Couldn't execute the instrumented binary $path_to_binary"
   return 1
 }
 
 function tear_down() {
   rm -rf coverage_srcs/
 }
 
+function run_coverage() {
+  (COVERAGE_DIR="$coverage_dir" COVERAGE_GCOV_PATH="$coverage_gcov_path" \
+   ROOT="$root" COVERAGE_MANIFEST="$coverage_manifest" \
+   COVERAGE_OUTPUT_FILE="$coverage_output_file" tools/test/collect_cc_coverage.sh)
```

Comment:
Abstract "tools/test/collect_cc_coverage.sh" to a constant.

Reply:
Done